Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs

```diff
@@ -607,23 +583,101 @@ void UpdateItems(NotifyCollectionChangedEventArgs e, int section, bool resetWhen
 				case NotifyCollectionChangedAction.Replace:
 					if (e.OldStartingIndex == -1 || groupReset)
 						goto case NotifyCollectionChangedAction.Reset;
-					Control.BeginUpdates();
-					Control.ReloadRows(GetPaths(section, e.OldStartingIndex, e.OldItems.Count), ReloadRowsAnimation);
-					Control.EndUpdates();
+
+					ReloadRows(e, section);
 
 					if (_estimatedRowHeight && e.OldStartingIndex == 0)
 						InvalidateCellCache();
 
-
 					break;
 
 				case NotifyCollectionChangedAction.Reset:
 					InvalidateCellCache();
-					Control.ReloadData();
+					ReloadData();
 					return;
 			}
 		}
 
+		void InsertRows(NotifyCollectionChangedEventArgs e, int section)
```

Comment:
can we pass just e.NewStartingIndex, e.NewItems.Count  ?

Reply:
Done.